src/aiokatcp/core.py

```diff
@@ -79,6 +79,8 @@ class Address:
     _IPV6_RE = re.compile(r"^\[(?P<host>[^]]+)\](:(?P<port>\d+))?$")
 
     def __init__(self, host: _IPAddress, port: Optional[int] = None) -> None:
+        if not isinstance(host, typing.get_args(_IPAddress)):
+            raise TypeError(f"{host} is not of either {typing.get_args(_IPAddress)}")
```

Comment:
Coveralls is failing and it looks like it's because there is no test coverage for this. Add a unit test that tries to construct an Address from the wrong type.